terraform/runbooks/redeploy_instances.md

```diff
@@ -0,0 +1,68 @@
+# How to Redeploy All Instances #
+
+Except when otherwise indicated, all commands listed below assume that
+the current working directory is `cyhy_amis/terraform`.
+
+* Force install all `ansible` and `packer/ansible` requirements:
+
+  ```bash
+  ansible-galaxy install --force -r ../ansible/requirements.yml
+  ansible-galaxy install --force -r ../packer/ansible/requirements.yml
+  ```
+
+* If any Nessus keys are being re-used, reset them via the
+  [Tenable Support Portal](https://support.tenable.com)
+
+* Notify all affected parties that downtime is starting
+
+* ssh to the `database1` instance and stop the `cyhy-commander` service:
+
+  ```bash
+  sudo systemctl stop cyhy-commander.service
+  ```
+
+* Terminate all EC2 instances through the
+  [AWS Console](https://aws.amazon.com/console/)
```

Comment:
What is the CLI command for this?

Reply:
It's `aws ec2 terminate-instances --instance-ids <id1> <id2> ...`  

Here's a handy snippet to get all of the instance IDs in a particular terraform workspace (assuming all of the instances are properly tagged with the workspace name):
`aws ec2 describe-instances --filters "Name=tag:Workspace,Values=<WORKSPACE_NAME>" --output text --query "Reservations[*].Instances[*].InstanceId"`

Our goal should be to not have to do this step at all.  Hopefully, after we move to terraform 0.12, we will be able to clean up our code so that `terraform apply` will be able to cleanly terminate our instances, then spin up their replacements (or in certain cases, we may be able create the new instances before destroying the old ones; eventually, we will want that model across the board).

Reply:
Roger.  Terraform 0.12 will fix all our woes!  ;)